sdcm/utils/validators/iotune.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from typing import TypedDict
+import yaml
+from argus.client.generic_result import ColumnMetadata, GenericResultTable, ResultType, Status, ValidationRule
+from argus.client.sct.client import ArgusSCTClient
+from sdcm.argus_results import submit_results_to_argus
+from sdcm.cluster import BaseNode
+from sdcm.remote.remote_file import remote_file
+from sdcm.sct_config import SCTConfiguration
+
+LOGGER = logging.getLogger(__name__)
+
+
+class IOTuneDiskParams(TypedDict):
+    mountpoint: str
+    read_iops: int
+    read_bandwidth: int
+    write_iops: int
+    write_bandwidth: int
+
+
+class IOProperties(TypedDict):
+    disks: list[IOTuneDiskParams]
+
+
+class IOTuneValidator:
+    def __init__(self, node: BaseNode, test_params: SCTConfiguration, argus_client: ArgusSCTClient = None):
+        self.node = node
+        self.params = test_params
+        self.node_io_properties = {}
+        self.argus_client = argus_client
+
+    def validate(self):
+        self._run_io_tune()
+        self._format_results_to_console()
+        self._submit_results_to_argus()
+
+    def _read_io_properties(self, io_props_path="/etc/scylla.d/io_properties.yaml") -> IOProperties:
+        with remote_file(self.node.remoter, io_props_path) as f:
+            return yaml.safe_load(f)
+
+    def _run_io_tune(self, temp_props_path="/tmp/io_properties.yaml") -> IOProperties:
+        self.node.remoter.sudo(f"iotune  --evaluation-directory /var/lib/scylla --properties-file {temp_props_path}")
+        self.node_io_properties = self._read_io_properties(temp_props_path)
+
+        return self.node_io_properties
+
+    def _format_results_to_console(self):
+        preset_properties = self._read_io_properties()
+        preset_disk = next(iter(preset_properties.get("disks")))
+
+        for disk in self.node_io_properties.get("disks"):
+            disk_copy = {**disk}
+            mountpoint = disk_copy.pop("mountpoint")
+            LOGGER.info("Disk performance values validation - testing %s", mountpoint)
+            for key, val in disk_copy.items():
+                preset_val = preset_disk.get(key)
+                diff = (val / preset_val - 1) * 100
+                LOGGER.info("[%s] %s: %s (%.0f)", mountpoint, key, val, diff)
+
+    def _submit_results_to_argus(self):
```

Comment:
this should be done outside of `IOTuneValidator` which should do only one thing: validating correctness of predefined io_properties against actual values: all things like submitting or printing in console could be done separately to not blur purpose of this class.

Reply:
okay, I'll move submission to the tester body.

Reply:
Moved to argus_results, kept the console output in the validator